### PR TITLE
New PS scripts for other APIs

### PR DESCRIPTION
**Pull request by markjbrown**
## DESCRIPTION

New PS scripts for all API's. Full create examples, List-Get in single script, update ru for db-level and container level. Created Common folder, copied scripts which are shared. 

After merge I will be removing a number of scripts in SQL folder and submitting a follow up PR to reduce the total number of scripts. All create examples will provision account and database.

Also adding @SnehaGunda for additional review.

## CHECKLIST

- [ ] This pull request was tested on:
  - [ ] PowerShell 5.1 on Windows
  - [X ] PowerShell 6.x on Windows
  - [ ] PowerShell 6.x on Linux
- [X ] Resources not created by the scripts in the pull request are __explicitly__ mentioned in a comment at the head of the file, __and__ have user-supplied variables or function arguments which correspond to the required resource identifiers.
- [X ] All user-supplied values are set in variables at the head of the file, after any comments.
- [ ] All passwords are user-supplied values, or generated by a secure random string generator.
- [X ] All Azure identifiers required to be universally unique are guaranteed to be so. 
- [X ] All scripts only use commands available in the latest release of the Azure PowerShell `Az` module. ([Command reference](https://docs.microsoft.com/en-us/powershell/module/?term=Az.))
  - [ ] I have an exception! (__optional__ - state for which service)
  - [ ] I'm requesting an exception! (__optional__ - you must include your MS alias for further discussion before PR review)
- [ ] All scripts contain only ASCII characters (no 'smart quotes' or other wide characters)

### CODE REVIEW

File: cosmosdb/common/ps-account-key-get.ps1 [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# List keys for an Azure Cosmos Account
```
Comment (SnehaGunda) [inline cosmosdb/common/ps-account-key-get.ps1]: Can you merge this script with get connection strings script?

File: cosmosdb/common/ps-account-key-regenerate.ps1 [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Regenerate the primary key for an Azure Cosmos Account
```
Comment (SnehaGunda) [inline cosmosdb/common/ps-account-key-regenerate.ps1]: we could merge this one too with the above script as its about keys?

File: cosmosdb/sql/ps-container-create.ps1 [hunk 3]
```diff
@@ -13,7 +13,7 @@ $ContainerProperties = @{
             "kind"="Hash"
         }
     }; 
-    "options"=@{ "Throughput"="400" }
+    "options"=@{ "Throughput"= 400 }
 } 
 New-AzResource -ResourceType "Microsoft.DocumentDb/databaseAccounts/apis/databases/containers" `
     -ApiVersion "2015-04-08" -ResourceGroupName $resourceGroupName `
```
Comment (SnehaGunda) [inline cosmosdb/sql/ps-container-create.ps1]: I guess the above 4 files will be deleted?

File: cosmosdb/cassandra/ps-cassandra-create.ps1 [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+$tableResourceName = $accountName + "/cassandra/" + $keyspaceName + "/" + $tableName
+
+# Create account
+$locations = @(
+    @{ "locationName"="West US 2"; "failoverPriority"=0 },
+    @{ "locationName"="East US 2"; "failoverPriority"=1 }
+)
+
+# Add an IP range filter to the properties
+# $ipRangeFilter = @{ "ipRangeFilter"="10.0.0.1" }
```
[the remark this answers is not in the document]
Comment (markjbrown) [reply cosmosdb/cassandra/ps-cassandra-create.ps1]: ok.

File: cosmosdb/cassandra/ps-cassandra-create.ps1 [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    "maxIntervalInSeconds"=300;
+    "maxStalenessPrefix"=100000
+}
+
+$accountProperties = @{
+    "capabilities"= @( @{ "name"="EnableCassandra" } );
+    "databaseAccountOfferType"="Standard";
+    "locations"=$locations;
+    # uncomment to add IP Firewall
+    # "ipRangeFilter"= $ipRangeFilter;
```
[the remark this answers is not in the document]
Comment (markjbrown) [reply cosmosdb/cassandra/ps-cassandra-create.ps1]: ok.

File: cosmosdb/cassandra/ps-cassandra-create.ps1 [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+
+# Create keyspace with shared throughput
+$keyspaceProperties = @{
+    "resource"=@{ "id"=$keyspaceName };
+    "options"=@{ "Throughput"= 400 }
+}
+New-AzResource -ResourceType "Microsoft.DocumentDb/databaseAccounts/apis/keyspaces" `
+    -ApiVersion "2015-04-08" -ResourceGroupName $resourceGroupName `
+    -Name $keyspaceResourceName -PropertyObject $keyspaceProperties
```
[the remark this answers is not in the document]
Comment (markjbrown) [reply cosmosdb/cassandra/ps-cassandra-create.ps1]: Yes, same pattern for all of these. Used in the resource property vs. KeyspaceResourceName is the interleaved path to the resource.

# Create keyspace with shared throughput
$keyspaceProperties = @{
    "resource"=@{ "id"=$keyspaceName };
    "options"=@{ "Throughput"= 400 }
}

File: cosmosdb/common/ps-account-key-get.ps1 (hunk 1 above)
Comment (markjbrown) [reply cosmosdb/common/ps-account-key-get.ps1]: done

File: cosmosdb/sql/ps-container-create.ps1 (hunk 3 above)
Comment (markjbrown) [reply cosmosdb/sql/ps-container-create.ps1]: I will keep a couple of these container-level scripts moving forward (will be fully self-contained) as they are common scenarios and are not highly intuitive initially to write. If I receive feedback or data suggesting some of the container-level scripts are not being used I will revisit.

Comment (markjbrown) [reply cosmosdb/sql/ps-container-create.ps1]: yes, they will be deleted.

**Comment by markjbrown** [comment]
@SnehaGunda  I have made all the recommended changes. Please review/merge. thanks.

**Comment by SnehaGunda** [comment]
@markjbrown looks good to me. 

@sptramer can you please merge

#sign-off

**Pull request merged**